amida-tech/amida-survey-microservice
# [DEVOPS-365] Slight (slight) improvement to Dockerfile.

**What's this PR do?**
Slightly tweaks the process to allow for correct deployments in Docker. Unfortunately, the survey microservice does not really build a production ready deployment right now. I've created [DEVOPS-366] to handle that. 

**Related JIRA tickets:**
DEVOPS-365, DEVOPS-366.

**How should this be manually tested?**
First, create a network if you haven't already:
`docker network create indaba-network`

Ensure that the docker instance has a database to work with. We're assuming the username and password in your .env are both set to amida, if not then please change. To start a docker instance of a database, run:
`docker run -p 5432:5432 --name survey-db --network indaba-network -e POSTGRES_USER=amida -e POSTGRES_PASSWORD=amida -e POSTGRES_DB=amida_survey_microservice postgres:9.6`

Then run a build in Docker with:
`docker build -t amida-survey-microservice .`

Finally try a deployment:
`docker run -p 9005:9005 --network indaba-network --name amida-survey-microservice -v <ABSOLUTE PATH>/.env:/app/.env -e SURVEY_SERVICE_PG_HOST=survey-db -e SURVEY_SERVICE_PG_USER=amida -e SURVEY_SERVICE_PG_PASSWORD=amida amida-survey-microservice yarn seed && yarn serve`

If you have any Postman requests, fire them at the endpoint to confirm completion.

**Any background context you want to provide?**
This blocks an ease-of-deployment for Indaba. I am currently trying to get four instances running for four people as simply as possible to cut down on... hour drain.

**Screenshots (if appropriate):**
